flink-table/flink-table-common/src/main/java/org/apache/flink/table/sources/PushDownTableSource.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.sources;
+
+/**
+ * Base interface for supporting operator push-down to a {@code TableSource}.
+ * A {@code TableSource} extending this interface will be able to acquire additional
+ * information regarding the push-down information of the table source.
+ *
+ * @param <T> The return type of the {@link TableSource}.
+ */
+public interface PushDownTableSource<T> {
+
+	/**
+	 * Describes the push down rule for this specific table source.
+	 *
+	 * @return A String explaining the {@link PushDownTableSource}.
+	 */
+	String explainPushDown();
```

Comment:
I figure I might be able to use some sort of default here but it will be hard without understanding what exactly is the pushdown is. open to any suggestion.

However I think having this API here alone prevents user from not overriding the `pushdown` and prevents the problematic pushdown rule application